libs/src/api/user.js

```diff
@@ -291,27 +301,39 @@ class Users extends Base {
     const newMetadata = this._cleanUserMetadata(metadata)
     this._validateUserMetadata(newMetadata)
 
+    // Error if libs instance does not already have existing user state
     const user = this.userStateManager.getCurrentUser()
     if (!user) {
       throw new Error('No current user')
     }
+
+    // Preserve old metadata object
     const oldMetadata = { ...user }
 
-    // Upload new metadata
-    const { metadataMultihash, metadataFileUUID } = await this.creatorNode.uploadCreatorContent(
-      newMetadata
-    )
-    // Update new metadata on chain
-    let updatedMultihashDecoded = Utils.decodeMultihash(metadataMultihash)
-    const { txReceipt } = await this.contracts.UserFactoryClient.updateMultihash(userId, updatedMultihashDecoded.digest)
-    const { latestBlockNumber } = await this._updateUserOperations(newMetadata, oldMetadata, userId)
+    // Update user creator_node_endpoint on chain if applicable
     if (newMetadata.creator_node_endpoint !== oldMetadata.creator_node_endpoint) {
-      await this._waitForCreatorNodeUpdate(newMetadata.user_id, newMetadata.creator_node_endpoint)
+      await this.contracts.UserFactoryClient.updateCreatorNodeEndpoint(userId, newMetadata['creator_node_endpoint'])
+
+      // Ensure DN has indexed creator_node_endpoint change
+      await this._waitForCreatorNodeEndpointIndexing(newMetadata.user_id, newMetadata.creator_node_endpoint)
     }
-    // Re-associate the user id with the metadata and block number
+
+    // Upload new metadata object to CN
+    const { metadataMultihash, metadataFileUUID } = await this.creatorNode.uploadCreatorContent(newMetadata)
```

Comment:
@SidSethi how would you feel about adding this check before this line as a safeguard to ensure that the value on chain == value indexed by discovery == value in libs?

```js
if (this.creatorNode.getEndpoint() !== newMetadata.creator_node_endpoint) {
  throw new Error(`Not connected to the correct creator node. Expected ${newMetadata.creator_node_endpoint}, got ${this.creatorNode.getEndpoint()}`)
}
```

Reply:
good call, added